### PR TITLE
Update addons-linter to 0.1.0

### LINKED ISSUE #1910: Forward listed/unlisted to addons-linter
The addons-linter currently doesn't handle anything in that regards but once it does, we should forward listed/unlisted flag.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
* Forward listed / unlisted to addons-linter (fixes #1910)
* Update tests to represent corrected id behavior in addons-linter

### CODE REVIEW

File: src/olympia/devhub/tests/test_tasks.py [hunk 1]
```diff
@@ -579,9 +579,9 @@ def test_amo_validator_addons_linter_error(self):
         self.create_switch('addons-linter')
 
         # This test assumes that `amo-validator` doesn't correctly
-        # validate a missing id in manifest.json
+        # validate a invalid id in manifest.json
         result = tasks.validate_file_path(
-            get_addon_file('invalid_webextension_missing_id.xpi'),
+            get_addon_file('invalid_webextension_invalid_id.xpi'),
```
Comment (tofumatt) [inline src/olympia/devhub/tests/test_tasks.py]: We validate IDs but don't require them. Should we at least file a bug to move this into the linter? (Or maybe it already does so)

**Comment by tofumatt** [comment]
r+wc

File: src/olympia/devhub/tests/test_tasks.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/devhub/tests/test_tasks.py]: The linter does everything correct. This test is based on the idea that `addons-linter` actually validates the id for a webextension and `amo-validator` doesn't so I'm using that to not just test "did I call this?" but also to verify that we get something expected back from the linter.

Comment (tofumatt) [inline src/olympia/devhub/tests/test_tasks.py]: ah okay cool. All good then!

**Pull request merged**